packages/@aws-cdk/aws-ec2/lib/user-data.ts

```diff
@@ -101,7 +260,19 @@ class CustomUserData extends UserData {
     this.lines.push(...commands);
   }
 
+  public addOnExitCommands(...commands: string[]): void {
+    this.onExitLines.push(...commands);
```

Comment:
You can't really pretend to support `onExitLines` if they're going to do something completely different. Better throw here as well. Please replace all the throws with more descriptive error messages (informing users about the likely mistake they made in THEIR code and what they should do to fix it), such as `CustomUserData does not support FooCommand, use UserData.forLinux() or UserData.forWindows() instead`.

Reply:
Updated